### PR TITLE
fix(pie-input): DSW-0 required input value

### PR DESCRIPTION
**Pull request by jamieomaguire**
---
"@justeattakeaway/pie-input": minor
---

[Added] - Make the `value` property required and default to an empty string

### CODE REVIEW

File: packages/components/pie-input/test/component/pie-input.spec.ts [hunk 1]
```diff
@@ -121,6 +121,21 @@ test.describe('PieInput - Component tests', () => {
...
+                    props: {
+                        value: undefined,
+                    } as InputProps,
+                });
+
+                // Act
+                const input = component.locator('input');
+
+                // Assert
+                expect((await input.inputValue())).toBe('');
```
Comment (xander-marjoram) [inline packages/components/pie-input/test/component/pie-input.spec.ts:136]: Out of interest would something along the lines of `page.inputValue('pie-input')` work, or do you need to find the inner input?

**Review by xander-marjoram — DISMISSED** [review]

File: packages/components/pie-input/test/component/pie-input.spec.ts (hunk 1 above)
Comment (jamieomaguire) [reply packages/components/pie-input/test/component/pie-input.spec.ts:136]: It's a nice suggestion. Unfortunately, Playwright throws on any element that isn't an input :

"Throws for non-input elements. However, if the element is inside the <label> element that has an associated control, returns the value of the control."

I would prefer to use an approach that doesn't check the underlying input though. I am unsure how we can do that. I considered something like:

`const inputValue = await page.evaluate(() => document.querySelector('pie-input')?.value);`

But it's cumbersome and didn't quite work either

**Review by fernandofranca — DISMISSED** [review]

**Review by ashleynolan — APPROVED** [review]

File: .changeset/unlucky-oranges-divide.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@justeattakeaway/pie-input": minor
+---
+
+[Added] - Make the `value` property required and default to an empty string
```
Comment (xander-marjoram) [inline .changeset/unlucky-oranges-divide.md:5]: If it's required, do we need a default?

Comment (jamieomaguire) [reply .changeset/unlucky-oranges-divide.md:5]: Screenshot 2024-01-25 at 17 25 29

**Review by fernandofranca — APPROVED** [review]

**Pull request merged**